### DISCUSSION TITLE
Build statically linked binary

### DISCUSSION THREAD

**Discussion by emcmillan** [mydumper/mydumper#1333] [Q&A] [answered]
If I build the binaries using "cmake .; make" as per the build instructions, the resulting binaries fail to run unless I set LD_LIBRARY_PATH to "/usr/lib64/mysql" AND I have the libmysqlclient.so.20 library in it.  However, the delivered binaries don't require the LD_LIBRARY_PATH setting, and are mach larger (10MB vs 256K).  How can I build the binaries like the delivered ones so they don't need LD_LIBRARY_PATH set?

Thanks

**Comment by davidducos** [comment]
In this file: https://github.com/mydumper/mydumper/blob/master/.circleci/config.yml you will find the CMAKED option like this:
```
        CMAKED: "-DMYSQL_LIBRARIES_mysqlclient:FILEPATH=/usr/lib64/mysql/libmysqlclient.a"
```
with the static library. Depending of the OS and version you will find it on different locations.

**Reply by emcmillan** [reply]
I must be doing this wrong.  Adding the flag to the _cmake_ command didn't help.  _cmake_ still says it's using the so file, the binary didn't change size, and it still requires the mysql client library before it will work.

`[root@a6105a0b1154 src]# cmake . -DMYSQL_LIBRARIES_mysqlclient:FILEPATH=/usr/lib64/mysql/libmysqlclient.a

-- Using mysql-config: /usr/bin/mysql_config

-- Found MySQL: /usr/include/mysql, /usr/lib64/mysql/libmysqlclient.so;/usr/lib64/libm.so;/usr/lib64/libatomic.so.1;/usr/lib/libcoredumper.a;/usr/lib64/libssl.so;/usr/lib64/libcrypto.so`

**Reply by davidducos** [reply, marked as answer]
did you `rm CMakeCache.txt` before cmake?
